### DISCUSSION TITLE
Gulp 4: How do I prevent from duplicated dependent task execution? #1392

### DISCUSSION THREAD

**Discussion by Spongman** [gulpjs/gulp#2742] [Help]
I have the exact same problem as described in #1392.

That issue was summarily closed without any indication of the solution.

**Comment by phated** [comment]
Questions need to be in Discussions. I'll move this for you.

**Reply by Spongman** [reply]
if there's somewhere in the docs that shows how to deal with multiple tasks having common dependencies, i'm not seeing it.

**Reply by Spongman** [reply]
oh. i see this: https://gulpjs.com/docs/en/api/parallel/#avoid-duplicating-tasks
that works for the trivial case, but it doesn't scale.
